src/include/daos/common.h

```diff
@@ -339,7 +339,7 @@ daos_sgl_buf_extend(d_sg_list_t *sgl, int idx, size_t new_size);
 	} while (0)
 /** Get the leftover space in an iov of sgl */
 #define daos_iov_left(sgl, iov_idx, iov_off)				\
-	((sgl)->sg_iovs[iov_idx].iov_len - (iov_off))
```

Comment:
Is this daos_iov_left() to calculate left data length or left buffer length? It looks to me some places used this function to calculate left data length, and 'iov_len' should be used if that's true?

Reply:
I agree it is kind of confuse
now for update we always set "iov_len == iov_buf_len == buffer_length", 
for fetch user is allowed to only set "iov_buf_len == buffer_length" and possibly set iov_len = 0.
So use iov_buf_len here seems fine? or you have better idea? need to differentiate fetch or update? (that seems not very convenient as well).

Reply:
I added some extra code for obj_update, that if iov_len != iov_buf_len then duplicate the sgl, as what we discussed, thx.